Remove timelines and APIv2

oliverguenther opened the pull request:
https://community.openproject.com/projects/openproject/work_packages/20938
https://community.openproject.com/projects/openproject/work_packages/24986

oliverguenther committed:
Remove APIV2 and project associations / reportings

[ci skip]

ulferts commented:
Just a quick note as I just stumbled over the code. The project dropdown seems to be using the api v2 https://github.com/opf/openproject/blob/dev/app/assets/javascripts/openproject.js#L91

ulferts committed:
salvage level_list from api/v2

We still need the functionality for the projects drop down. Thus we reuse the production hardened code from api/v2

ulferts commented:
Salvaged the level_list functionality from v2 and placed it in the `ProjectsController`.

ulferts commented:
legacy specs in https://github.com/opf/openproject/tree/dev/spec_legacy/integration/api_spec will probably have to be deleted

ulferts commented:
We should be able to also remove the `PlanningElementsHelper` module

oliverguenther committed:
Remove timelines

oliverguenther committed:
Fix removed methods from timelines helper

oliverguenther committed:
Remove unused JS factories

oliverguenther committed:
Add color helper

oliverguenther committed:
Remove timeline hook for user

oliverguenther committed:
Remove reporting specs

oliverguenther committed:
Remove timeline reference in copy

oliverguenther committed:
Adapt level_json path

oliverguenther committed:
Fix specs

oliverguenther committed:
Removed unused select2 code

ulferts committed:
remove planning_element from permitted_params

oliverguenther committed:
Re-add empty helpers object

ulferts committed:
cleanup timelines data

oliverguenther committed:
Fix escape in member select2

oliverguenther committed:
Respond 410 gone to all requests under /api/v2*

oliverguenther committed:
Remove legacy tests

[ci skip]
